src/pages/workspaces/workspace/RequestAccessModal.js

```diff
@@ -0,0 +1,107 @@
+import { useState } from 'react'
+import { div, h, table, tbody, td, th, thead, tr } from 'react-hyperscript-helpers'
+import { buttonPrimary, link } from 'src/components/common'
+import { centeredSpinner, icon } from 'src/components/icons'
+import Modal from 'src/components/Modal'
+import { Ajax, useCancellation } from 'src/libs/ajax'
+import { withErrorReporting } from 'src/libs/error'
+import { cond, useOnMount, withBusyState } from 'src/libs/utils'
+
+
+export const RequestAccessModal = ({ onDismiss, workspace }) => {
+  const [groups, setGroups] = useState([])
+  const [accessInstructions, setAccessInstructions] = useState([])
+  const [loading, setLoading] = useState(false)
+  const signal = useCancellation()
+
+  const { Groups, Workspaces } = Ajax(signal)
+
+  const fetchGroups = withErrorReporting('Error loading groups')(async () => {
+    setGroups(await Groups.list())
+  })
+
+  const fetchAccessInstructions = withErrorReporting('Error loading instructions')(async () => {
+    setAccessInstructions(await Workspaces.workspace(workspace.workspace.namespace, workspace.workspace.name).accessInstructions())
+  })
+
+  const fetchAll = withBusyState(setLoading)(async () => {
+    await Promise.all([
+      fetchGroups(),
+      fetchAccessInstructions()
+    ])
+  })
+
+  useOnMount(() => {
+    fetchAll()
+  })
+
+  const groupNames = _.map('groupName', groups)
+  const authDomain = workspace.workspace.authorizationDomain
+  return h(Modal, {
+    title: 'Request Access',
+    width: '40rem',
+    showCancel: false,
+    onDismiss
+  }, [
+    div([`
+      You cannot access this workspace because it is protected by an Authorization Domain.
+      You need to obtain permission from an admin of each group in the Authorization Domain in order to get access.
+      Clicking the "Request Access" button below will send an email to the admins of that group.`]),
+    div({ style: { marginTop: '1rem' } }, [
+      link({
+        href: 'https://support.terra.bio/hc/en-us/articles/360026775691'
+      }, ['Learn more about Authorization Domains', icon('pop-out', { size: 12, style: { marginLeft: '0.25rem' } })])
+    ]),
+    loading ?
+      centeredSpinner({ size: 32 }) :
+      table({ style: { margin: '1rem', width: '100%' } }, [
+        thead([
+          tr({ style: { height: '2rem' } }, [th({ style: { textAlign: 'left' } }, ['Group Name']), th({ style: { textAlign: 'left', width: '15rem' } }, ['Access'])])
+        ]),
+        tbody(
+          _.map(({ membersGroupName: groupName }) => tr({ style: { height: '2rem' } }, [
+            td([groupName]),
+            td([
+              _.includes(groupName, groupNames) ?
+                'Yes' :
+                h(RequestAccessButton, {
+                  groupName,
+                  instructions: accessInstructions[groupName]
+                })
+            ])
+          ]), authDomain)
+        )
+      ])
+  ])
+}
+
+const RequestAccessButton = ({ groupName, instructions }) => {
```

Comment:
TOL: It's a bit confusing that `instructions` isn't used. If this feature isn't imminent, I wonder if we should just omit that code path until we actually need it.

Reply:
I'd like to give it a little time. My feeling is that we should be fixing this regression pretty quickly. Furthermore, if I recursively do away with all of the unused symbols, the previous refactor of using `Promise.all` to fetch the instructions in parallel with fetching the groups. I know it's not a ton of work but I'd rather leave it in rather than be too aggressive killing it.